sqlserver/datadog_checks/sqlserver/metrics.py

```diff
@@ -561,6 +561,128 @@ def fetch_metric(self, rows, columns):
             self.report_function(metric_name, column_val, tags=metric_tags)
 
 
+class SqlVirtualFileIOStats(BaseSqlServerMetric):
+    CUSTOM_QUERIES_AVAILABLE = False
+
+    # Default base query works for server-wide view of all master files
+    QUERY_MASTER_FILES = """
+        SELECT
+            DB_NAME(fs.database_id) AS database_name,
+            mf.state_desc AS state_desc,
+            mf.name AS logical_name,
+            mf.physical_name AS physical_name,
+            fs.num_of_reads AS num_of_reads,
+            fs.num_of_bytes_read AS num_of_bytes_read,
+            fs.io_stall_read_ms AS io_stall_read_ms,
+            fs.io_stall_queued_read_ms AS io_stall_queued_read_ms,
+            fs.num_of_writes AS num_of_writes,
+            fs.num_of_bytes_written AS num_of_bytes_written,
+            fs.io_stall_write_ms AS io_stall_write_ms,
+            fs.io_stall_queued_write_ms AS io_stall_queued_write_ms,
```

Comment:
looks like this column isn't supported for sqlserver 2008-2012? https://docs.microsoft.com/en-us/sql/relational-databases/system-dynamic-management-views/sys-dm-io-virtual-file-stats-transact-sql?view=sql-server-ver15

Reply:
Ah interesting. AFAIK we are not planning to support 2008-2012 out of the gate so it may not be an issue. We can always revisit when we do add support.

Reply:
OK, might be worth a comment in case we get a question about it. We are planning on supporting these older versions eventually right?

Reply:
I don't know if we need to explicitly call it out anywhere. With support, we will add those versions to our test matrix and tests will fail. But for now, these metrics will just be skipped and the rest of the integration will continue to run fine.